rectools/models/implicit_als.py

```diff
@@ -172,7 +178,20 @@ def fit_als_with_features_separately_inplace(
     verbose : int
          Whether to print output.
     """
+    # If model was fitted we should drop any learnt embeddings except actual latent factors
+    if model.user_factors is not None and model.item_factors is not None:
+        if isinstance(model, GPUAlternatingLeastSquares):  # pragma: no cover
+            # Without .copy() gpu.Matrix will break correct slicing
+            user_factors = get_users_vectors(model)[:, : model.factors].copy()
+            item_factors = get_items_vectors(model)[:, : model.factors].copy()
+            model.user_factors = implicit.gpu.Matrix(user_factors)
+            model.item_factors = implicit.gpu.Matrix(item_factors)
+        else:
+            model.user_factors = model.user_factors[:, : model.factors]
```

Comment:
I did `_set_factors` but in the way where I can reuse it in another place to stop duplicating code. So paer of the logic is still in the main func

Also can't do `get_factors(model)` with Tuple output because we sometimes need only user or only item factors.